src/osx/cocoa/textctrl.mm

```diff
@@ -1018,6 +1062,25 @@ - (void)controlTextDidEndEditing:(NSNotification *)aNotification
     wxString st(wxMacConvertNewlines10To13(str));
     wxMacEditHelper helper(m_textView);
     wxWidgetCocoaNativeKeyDownSuspender suspend(this);
+
+    wxWidgetCocoaImpl* impl = (wxWidgetCocoaImpl* ) wxWidgetImpl::FindFromWXWidget( m_textView );
+    if ( impl )
+    {
+        wxTextCtrl *textCtrl = (wxTextCtrl *) impl->GetWXPeer();
+        if( textCtrl && textCtrl->GetTextPeer() )
+        {
+            if( textCtrl->HasFlag( wxTE_MULTILINE ) )
+            {
+                auto maxlen = textCtrl->GetMaxLen();
+                if( textCtrl->GetValue().length() + st.length() > maxlen )
```

Comment:
Overflow also seems to be possible here. Comparing `st.length()` with `maxlen` first would make it safe.